doc/tools/make_rst.py

```diff
@@ -587,12 +592,26 @@ def make_rst_class(class_def, state, dry_run, output_dir):  # type: (ClassDef, S
     else:
         f = open(os.path.join(output_dir, "class_" + class_name.lower() + ".rst"), "w", encoding="utf-8")
 
-    # Warn contributors not to edit this file directly
+    # Remove the "Edit on Github" button from the online docs page.
     f.write(":github_url: hide\n\n")
-    f.write(".. Generated automatically by doc/tools/make_rst.py in Godot's source tree.\n")
-    f.write(".. DO NOT EDIT THIS FILE, but the " + class_name + ".xml source instead.\n")
-    f.write(".. The source is found in doc/classes or modules/<name>/doc_classes.\n\n")
 
+    # Warn contributors not to edit this file directly.
+    # Also provide links to the source files for reference.
+
+    git_branch = "master"
+    if hasattr(version, "docs") and version.docs != "latest":
+        git_branch = version.docs
+
+    source_xml_path = os.path.normpath(os.path.join("doc", class_def.filepath)).replace("\\", "/")
```

Comment:
Doesn't work on Linux:
```
.. XML source: https://github.com/godotengine/godot/tree/master//home/akien/Projects/godot/godot.git/doc/classes/AESContext.xml.
```

Reply:
Huh, didn't expect the path not to be relative. Let me investigate it a bit more.

Reply:
Try `os.path.relpath` maybe?

Reply:
This works for me:
```diff
diff --git a/doc/tools/make_rst.py b/doc/tools/make_rst.py
index a56e4b338e..12d153a7aa 100755
--- a/doc/tools/make_rst.py
+++ b/doc/tools/make_rst.py
@@ -602,7 +602,7 @@ def make_rst_class(class_def, state, dry_run, output_dir):  # type: (ClassDef, S
     if hasattr(version, "docs") and version.docs != "latest":
         git_branch = version.docs
 
-    source_xml_path = os.path.normpath(os.path.join("doc", class_def.filepath)).replace("\\", "/")
+    source_xml_path = os.path.relpath(class_def.filepath, script_directory + "/../..").replace("\\", "/")
     source_github_url = "https://github.com/godotengine/godot/tree/{}/{}".format(git_branch, source_xml_path)
     generator_github_url = "https://github.com/godotengine/godot/tree/{}/doc/tools/make_rst.py".format(git_branch)
 
```

Reply:
Fix in https://github.com/godotengine/godot/pull/63530